app/models/menu_items_menu.rb

```diff
@@ -4,7 +4,6 @@ class MenuItemsMenu < ActiveRecord::Base
 
   validates :daily_stock, presence: true, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
   validates :menu_item, presence: true
-  validates :menu, presence: true
```

Comment:
To be honest, I don't understand your comment. I don't see how the above flow can have anything to do with what happens in production. Of course you'll have a `nil` id on `m` if you use the `FactoryGirl.build` method. I'm not even sure you could call `m.menu_items << m` on an object that is to persistent.

Anyway, I have nothing against removing this validation but I feel it's a little hacky. The problem must lie in the way we are adding the associated menu_items in AA.